Automattic/wp-calypso
# Video Onboarding: Add video upload celebration modal 🎉

#### Proposed Changes

* Displays a celebration modal after a video is uploaded to the editor.
* Only displays is the site signed up via the video onboarding flow.
* **Heavily** inspired by the existing `seller celebration modal` (basically a copy with some tweaks!)

<img width="665" alt="Screen Shot 2022-10-25 at 3 10 54 PM" src="https://user-images.githubusercontent.com/789137/197893720-e40ddab3-7334-4681-b4eb-9e2292dce181.png">

#### Testing Instructions

* You might want to edit line `105` of `apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/video-celebration-modal/index.jsx`. Change the `https://wordpress.com` to `http://calypso.localhost:3000`. This way you can test going back to the launchpad from the modal locally.
* Ensure you have a `wpcom-sandbox` ssh alias in your `.ssh/config` file.
* From your local checkout of calypso, run `yarn dev --sync` to sync the editing toolkit changes to your sandbox.
* Create a new site at `http://calypso.localhost:3000/start/intro?flow=videopress`.
* Go all the way through until you get to the launchpad. Copy the url for your new site, and quit your browser.
* Add the url to your hosts file to sandbox the new site.
* Relaunch your browser and return to the launchpad. Click `Upload your first video`.
* When at the site editor, make a simple text change and click save. You should see no modal because you haven't uploaded a video yet.
* Upload a video, then save. You should see the modal! Verify the buttons work (if you dismiss or click `continue editing` you won't see the modal again so maybe test the `Back to setup` button first).
* Reload the page, make another small change and save. You should _not_ see the modal because you have already seen it once.

#### Pre-merge Checklist
